File: iguana-wk1-app-6_1_2/web_docs/templates/edit_group.cs
<link rel="stylesheet" type="text/css" href="<?cs var:skin("transition.css") ?>" />
<style type="text/css">
a.button {
   float:left;
}
</style>
   <div id="iguana">
      <div id="cookie_crumb">
         <a href="/settings">Settings</a> &gt; <a href="/settings#Page=roles">Roles</a> &gt; <?cs if:Role.Name ?>Edit<?cs else ?>Add<?cs /if ?> Role
      </div>

      <div id="dashboard_body">
         <center>
            <form name="role_data" id="role_data">
               <div class="workarea" style="width: 640px;">

                  <input id="current_user_can_admin" type="hidden" value="<?cs var:CurrentUserCanAdmin ?>" />
                  <input id="current_user" type="hidden" value="<?cs var:html_escape(CurrentUser) ?>" />

                  <h1>
                     <?cs if:Role.Name ?>
                     Edit Role
                     <?cs else ?>
                     Add Role
                     <?cs /if ?>
                  </h1>

                  <div class="border_green"></div>

                  <div id="result" class="<?cs if:ErrorMessage ?>error<?cs elif:StatusMessage ?>success<?cs else?>indeterminate<?cs /if ?>">

                     <div class="result_buttons_system">
                        <a id="result_close" ><img src="images/close_button.gif"/></a>
                     </div>

                     <div id="result_title" class="<?cs if:ErrorMessage ?>error<?cs elif:StatusMessage ?>success<?cs /if ?>">

                        <?cs if:ErrorMessage ?>

                        Error
                        <?cs set:Content=ErrorMessage ?>

                        <?cs else ?>
<?cs include:"browse_macro.cs" ?>
                        Success
                        <?cs set:Content=StatusMessage ?>

                        <?cs /if ?>

                     </div>

                     <div id="result_content">
                        <?cs var:html_escape(Content) ?>
                     </div>
                  </div>

                  <div id="group_info" >
                     <div class="detail_header">
                        <h2>
                        Role Details
                        </h2>
                     </div>
                     <fieldset>
                        <ul>
                           <li <?cs if:!Role.Name ?>class="generic_control"<?cs /if ?>>
                              <div class="label_text">
                              Name
                              </div>

                              <?cs if:Role.Name ?>

                              <div class="value_text"><?cs var:html_escape(Role.Name) ?>
                              </div>

                              <input type="hidden" id="role" name="role" value="<?cs var:html_escape(Role.Name) ?>" />
                              <input type="hidden" id="action" name="action" value="editRole" />

                              <?cs else ?>

                              <div class="field_text">
                                 <input type="text" class="text" id="role" name="role" maxlength="30" />
                              </div>
                              <input type="hidden" id="action" name="action" value="createRole" />

                              <?cs /if ?>

                           </li>
                           <li class="textarea_control">
                              <div class="label_text">
                              Description
                              </div>
                              <div class="field_text">
                                 <textarea class="group_description" id="description" name="description"><?cs var:html_escape(Role.Description) ?></textarea>
                              </div>
                           </li>
                        </ul>
                     </fieldset>
                  </div>

                  <?cs if:CurrentUserCanAdmin ?>

                  <div id="permissions_border" class="border_grey" ></div>
                  <div id="permissions" >
                     <div class="detail_header">
                     <h2>
                     Channel Groups
                     </h2>
                     </div>

                     <div id="permissions_nonadmin">
                        <fieldset>
                           <ul>
                              <li class="permissions_control">

                                 <?cs if:Role.Name == "Administrators" ?>
                                 <span>The Administrators role has full access to all features in Iguana.  This cannot be modified.</span>
                                 <?cs else ?>

                                 <div id="role_channel_group_list" class="list" >

                                    <?cs if:ChannelGroupCount == 0 ?>
                                       <span>There are no configured channel groups.</span>
                                    <?cs else ?>

                                    <input id="channel_group_count" type="hidden" value="<?cs var:#ChannelGroupCount ?>" />

                                    <div class="permission_row_select_all">

                                       <div class="channel-group-dropdown" >
                                          <div id="group_drop_down_cell" <?cs if:ViewableChannelGroupCount == ChannelGroupCount ?>class="hidden_not_place_holder"<?cs /if ?>>
                                             <select id="group_drop_down">
                                                <option value="0">Add a channel group</option>
                                                <?cs each:item = ChannelGroup ?>
                                                   <?cs if:!item.Viewable ?>
                                                      <option value="<?cs var:item.ID ?>"><?cs var:item.Name ?></option>
                                                   <?cs /if ?>
                                                <?cs /each ?>
                                             </select>
                                          </div>
                                       </div>

                                       <div id="select_all" <?cs if:!ViewableChannelGroupCount ?>class="hidden_not_place_holder"<?cs /if ?> >
                                          <div id="select_all_reconfigure" class="reconfigurable" >
                                             <a id="reconfigure_all"><em> All</em></a> | <a id="reconfigure_none"><em> None</em></a>
                                          </div>
                                          <div id="select_all_startstop" class="startstopenabled label" >
                                             <a id="startstop_all"><em> All</em></a> | <a id="startstop_none"><em> None</em></a>
                                          </div>
                                          <div id="select_all_export" class="exportenabled label" >
                                             <a id="exportlogs_all"><em> All</em></a> | <a id="exportlogs_none"><em> None</em></a>
                                          </div>
                                          <div id="select_all_view_logs" class="view_logs_enabled label" >
                                             <a id="view_logs_all"><em> All</em></a> | <a id="view_logs_none"><em> None</em></a>
                                          </div>

                                          <div id="select_all_translator_ide" class="translator_ide_enabled label" >
                                             <a id="translator_ide_all"><em> All</em></a> | <a id="translator_ide_none"><em> None</em></a>
                                          </div>
                                       </div>
                                    </div>
                                    <ul>

                                       <?cs each:item = ChannelGroup ?>

                                       <li id="channel_group_<?cs var:item.ID ?>" class="channel_group_row dark_row <?cs if:!item.Viewable ?>hidden_not_place_holder<?cs /if ?>">
                                          <div class="permission_row" >
                                             <input class="group_id" type="hidden" value="<?cs var:item.ID ?>" />
                                             <div class="viewable">
                                                <a class="label channel_group_toggle">
                                                   <input class="group_id" type="hidden" value="<?cs var:item.ID ?>" />
                                                   <img id="channel_group_details_<?cs var:item.ID ?>_toggle" src="/images/arrow-contracted.gif" />
                                                   <span id="channel_group_name_<?cs var:item.ID ?>" ><?cs var:html_escape(item.Name) ?></span>
                                                </a>

                                                <a class="channel_group_remove">
                                                   <input class="group_id" type="hidden" value="<?cs var:item.ID ?>" />
                                                   <input class="group_name" type="hidden" value="<?cs var:html_escape(item.Name) ?>" />
                                                   [remove]
                                                </a>

                                             </div>
                                             <div class="reconfigurable label_checkbox" id="channel_group_reconfigure_<?cs var:item.ID ?>_cell" >
                                                <input type="checkbox" id="channel_group_reconfigure_<?cs var:item.ID ?>" name="channel_group_reconfigure_<?cs var:item.ID ?>" <?cs if:item.Reconfigurable ?>checked<?cs /if ?> class="checkbox" />
                                                <label for="channel_group_reconfigure_<?cs var:item.ID ?>">Edit</label>
                                             </div>
                                             <div class="startstopenabled label_checkbox" id="channel_group_startstop_<?cs var:item.ID ?>_cell" >
                                                <input type="checkbox" id="channel_group_startstop_<?cs var:item.ID ?>" name="channel_group_startstop_<?cs var:item.ID ?>" <?cs if:item.StartStopEnabled ?>checked<?cs /if ?> class="checkbox" />
                                                <label for="channel_group_startstop_<?cs var:item.ID ?>">Start/Stop</label>
                                             </div>
                                             <div class="exportenabled label_checkbox" id="channel_group_export_<?cs var:item.ID ?>_cell" >
                                                <input type="checkbox" id="channel_group_export_<?cs var:item.ID ?>" name="channel_group_export_<?cs var:item.ID ?>" <?cs if:item.ExportLogEnabled ?>checked<?cs /if ?> class="checkbox" />
                                                <label for="channel_group_export_<?cs var:item.ID ?>">Export Logs</label>
                                             </div>

                                             <?cs # View Logs is enabled by default for new roles. ?>
                                             <div class="view_logs_enabled label_checkbox" id="channel_group_view_logs_<?cs var:item.ID ?>_cell" >
                                                <input type="checkbox" id="channel_group_view_logs_<?cs var:item.ID ?>" name="channel_group_view_logs_<?cs var:item.ID ?>" <?cs if:item.ViewLogsEnabled || !Role.Name ?>checked<?cs /if ?> class="checkbox" />
                                                <label for="channel_group_view_logs_<?cs var:item.ID ?>">View Logs</label>
                                             </div>
                                             <div class="translator_ide_enabled label_checkbox" id="channel_group_translator_ide_<?cs var:item.ID ?>_cell" >
                                                <input type="checkbox" id="channel_group_translator_ide_<?cs var:item.ID ?>" name="channel_group_translator_ide_<?cs var:item.ID ?>" <?cs if:item.TranslatorIDEEnabled ?>checked<?cs /if ?> class="checkbox" />
                                                <label for="channel_group_translator_ide_<?cs var:item.ID ?>">Translator IDE</label>
                                             </div>
                                          </div>
                                          <div id="channel_group_details_<?cs var:item.ID ?>" class="permissions_row_details hidden_not_place_holder" >
                                             <input class="group_id" type="hidden" value="<?cs var:item.ID ?>" />
                                             <div style="padding-left:10px;" >

                                                <?cs if:item.ChannelCount == #0 ?>

                                                There are no channels in this group.

                                                <?cs else ?>

                                                <ul>

                                                   <?cs each:item_channel = item.Channels ?>

                                                   <li>
                                                      <?cs var:html_escape(item_channel.Name) ?>
                                                   </li>

                                                   <?cs /each ?>

                                                </ul>

                                                <?cs /if ?>

                                             </div>
                                          </div>

                                       </li>

                                       <?cs /each ?>

                                    </ul>

                                    <?cs /if ?>

                                 </div>

                                 <?cs /if ?>

                              </li>
                           </ul>

                           <?cs each:item = ChannelGroup ?>
                           <input id="<?cs var:item.ID ?>" name="<?cs var:item.ID ?>" class="channel_group_view_state" type="hidden" value="<?cs if:!item.Viewable ?>invisible<?cs else ?>visible<?cs /if ?>" />
                           <?cs /each ?>

                           <div id="viewable_channel_group_none_blurb" class="effective_permissions_blurb">
                           There are no viewable channel groups for this role.
                           </div>
                        </fieldset>
                     </div>
                  </div>

                  <?cs /if ?>

                  <div class="border_grey" ></div>
                  <div id="buttons">

                           <?cs if:Role.Name ?>

                           <input type="hidden" id="apply_changes" name="apply_changes" value="Apply Changes">
                           <a class="action-button blue" id="SaveChanges">Save Changes</a>

                           <?cs else ?>

                           <input type="hidden" id="apply_changes" name="apply_changes" value="Add Role">
                           <a class="action-button blue" id="SaveChanges">Add Role</a>
                           <?cs /if ?>

                              <a class="action-button blue" href="/settings#Page=roles">Cancel</a>
                  </div>
               </div>
            </form>
         </center>
      </div>
   </div>
</div>

<div id="side_panel">

   <div id="side_table">

      <div id="side_header">
      Page Help
      </div>

      <div id="side_body">

         <h4 class="side_title">Overview</h4>

         <?cs if:Role.Name ?>

         <p>
         On this page, you can change the permissions for the role.
         </p>

         <?cs else ?>

         <p>
         Use this page to add a new role to this Iguana server.
         </p>

         <?cs /if ?>

	 <h4 class="side_title">Related Settings</h4>

         <a href="/settings#Page=email_status">Email Notification</a><br/>
         <a href="/settings#Page=channel/group">Channel Groups</a>

      </div>

      <div class="side_item">

         <h4 class="side_title">Help Links</h4>

         <ul class="help_link_icon">

            <?cs if:Role.Name ?>

            <li>
               <a href="<?cs var:help_link('iguana4_assigning_user_to_group') ?>" target="_blank">Adding a User to a Role</a>
            </li>
            <li>
               <a href="<?cs var:help_link('iguana4_user_group_permissions') ?>" target="_blank">Editing Role Permissions</a>
            </li>

            <?cs else ?>

            <li>
               <a href="<?cs var:help_link('iguana4_creating_user_group') ?>" target="_blank">Creating a Role</a>
            </li>

            <?cs /if ?>

         </ul>

      </div>

   </div>

<script type="text/javascript" src="<?cs var:iguana_version_js("/templates/edit_group.js") ?>"></script>
<script type="text/javascript">
$(document).ready(function() {
   var ChannelGroupNames = new Array(<?cs var:#ChannelGroupCount ?>);
   <?cs set:count = #0 ?>
   <?cs each: group = ChannelGroup ?>
   ChannelGroupNames[<?cs var:count ?>] = "<?cs var:js_escape(group.Name) ?>";
   <?cs set:count = count + #1 ?>
   <?cs /each ?>

   $("form#role_data").submit({uri: '/roles', form_id: "role_data"}, SettingsHelpers.submitForm);
   $("a#SaveChanges").click(function(event) {
      event.preventDefault();
      if (! update_role()) { return false; }
      $("form#role_data").submit();
   });

});

</script>

File: iguana-wk2-app-6_1_2/web_docs/templates/edit_group.js
/** @license
 * Copyright (c) 2010-2011 iNTERFACEWARE Inc.  All rights reserved.
 */

// Javascript routines for edit_group.cs page

function update_role()
{ 
   if("" == $("#role").val())
   {
      showErrorMessage("Roles require a name.");

      document.role_data.role.focus();
      return false;
   }
   else
   {
      if("createRole" == $("#action").val())
      {
         var Result = validateRoleName($("#role").val());

         if(!Result)
         {
            return false;
         }
      }
   } 

   return true;
}

// Selects or deselects all permission of a particular type (i.e. view, reconfigure, startstop) 
// for a given accessor type (i.e. user or role).
function onToggleAllPermissionBoxes(ResourceType, PermissionType, State)
{
   // Toggle the checkboxes within the selected column. 
   for (var i = 0; i < document.role_data.elements.length; i++) 
   {
      var e = document.role_data.elements[i];

      if((e.name != undefined || e.name != null) && 
          (e.name.match("^" + ResourceType + "_" + PermissionType) == ResourceType + "_" + PermissionType) &&
          (e.type == "checkbox") && 
          (e.className == "checkbox")) 
      {
         e.checked = State;
      }
   }

   // If we're dealing with a global select of the view column, then we will have to adjust 
   // the other two columns.
   if("view" == PermissionType)
   { 
      for (var i = 0; i < document.role_data.elements.length; i++) 
      {
         var e = document.role_data.elements[i];

         if((e.name != undefined || e.name != null) && 
          (e.name.match("^" + ResourceType + "_view") == ResourceType + "_view")) 
         {
            var e_reconfigure = document.getElementById(e.id.replace("view", "reconfigure"));
            var e_startstop = document.getElementById(e.id.replace("view", "startstop"));

            if(!e.checked)
            {
               e_reconfigure.checked = false;
               e_reconfigure.className = "checkbox_disabled";
               e_reconfigure.onclick = function(){ return false; };

               e_startstop.checked = false;
               e_startstop.className = "checkbox_disabled";
               e_startstop.onclick = function(){ return false; };
            }
            else
            {
               if(e.className == "checkbox")
               {
                  e_reconfigure.className = "checkbox";
                  e_reconfigure.onclick = function(){ return true; };

                  e_startstop.className = "checkbox";
                  e_startstop.onclick = function(){ return true; };
               }
            }
         }
      }
   }
}

function validateRoleName(Name)
{
   var EncodedName = encodeURIComponent(Name);

   var AJAXCommandName = "validate_user_group_name";
   var AJAXVariableName = "Name=";

   var Result = true;

   $.ajax({url: AJAXCommandName, data: AJAXVariableName + EncodedName,
      async: false, 
      success: 
      function(data) 
      {                        
         try
         {
            var Response = data;
 
            if(Response) 
            {
               Result = Response.Result;

               if(!Result)
               {
                  showErrorMessage(Response.ErrorMessage);
               }
            }
         }
         catch(err)
         {                             
         }        
      },

      error:
      function()
      {           
         MiniLogin.show('Iguana is not Responding', function() { validateRoleName(Name) });
      }
   });

   return Result;
}

function stripChannelGroupRows()
{
   $("li.channel_group_row:visible").each( function(index)
                                           { 
                                              if(index % 2)
                                              { 
                                                 $(this).removeClass("dark_row").addClass("light_row"); 
                                              }
                                              else
                                              { 
                                                 $(this).removeClass("light_row").addClass("dark_row");
                                              } 
                                           } );
}

// Processes selections made in the viewable channel groups drop down.
function onDropdownChange(Dropdown)
{
   if(!Dropdown || Dropdown.value == "0")
   {
      return;
   }
 
   $("#channel_group_" + Dropdown.value).show();  
   $("#" + Dropdown.value).val("visible");  

   $("#select_all").show();

   for (i = Dropdown.length - 1; i >= 1; i--) 
   {
      if (Dropdown.options[i].selected) 
      {
         Dropdown.remove(i);
         break;
      }
   }      
   
   if(Dropdown.length < 2)
   {
      $("#group_drop_down_cell").hide();
   } 

   Dropdown.value = "0";

   stripChannelGroupRows();

   $("#viewable_channel_group_none_blurb").hide();
}

// Processes clicks on roles remove buttons.
function onRemoveGroup(ID, Name)
{
   var DropDown = $("#group_drop_down")[0];

   $("#channel_group_" + ID).hide();
   $("#" + ID).val("invisible"); 

   var Option = document.createElement('option');
   Option.text = Name;
   Option.value = ID;
   
   var InsertIndex = 1;

   if(DropDown.options.length > 1)
   {
      for (i = 0; i < ChannelGroupNames.length; ++i) 
      {
         if(ChannelGroupNames[i] == Name)
         {
            break;
         }

         if (DropDown.options[InsertIndex].text == ChannelGroupNames[i])
         {
            ++InsertIndex;
         }
      }
   }      

   try {
     DropDown.options.add(Option, InsertIndex); // standards compliant; doesn't work in IE
   }
   catch(ex) {
     DropDown.options.add(Option, DropDown.selectedIndex); // IE only
   }

   if(DropDown.options.length == parseInt($("#channel_group_count").val()) + 1)
   {
      $("#select_all").hide();
   }

   $("#group_drop_down_cell").show();

   stripChannelGroupRows();

   if(!$(".channel_group_view_state[value=visible]").length)
   {
      $("#viewable_channel_group_none_blurb").show();
   }
   else
   {
      $("#viewable_channel_group_none_blurb").hide();
   }
}

// Changes the row color on a mouseover.
function onRowEntered(Row)
{
   Row.removeClass("dark_row").removeClass("light_row").addClass("dark_row_highlighted");
}

// Changes the row color on a mouseexit.
function onRowExited()
{
   $("li.channel_group_row").removeClass("dark_row_highlighted").addClass("dark_row");
   stripChannelGroupRows();
}

$(document).ready(function() 
{
   $("#group_drop_down").bind("change", function(){ onDropdownChange($(this)[0]); } );

   if("Administrators" == $("#role").val() && "editRole" == $("#action").val())
   {
      $("#group_drop_down_cell").hide();
   }

   $("a.channel_group_toggle").bind("click", function(){ onListViewToggle("channel_group_details_" + $(".group_id", this).val()); } );

   $("a.channel_group_remove").bind("click", function(){ onRemoveGroup($(".group_id", this).val(), $(".group_name", this).val()); } );

   $("#reconfigure_all").attr("href", "javascript:onToggleAllPermissionBoxes('channel_group', 'reconfigure', true);");
   $("#startstop_all").attr("href", "javascript:onToggleAllPermissionBoxes('channel_group', 'startstop', true);");
   $("#exportlogs_all").attr("href", "javascript:onToggleAllPermissionBoxes('channel_group', 'export', true);");
   $("#view_logs_all").attr("href", "javascript:onToggleAllPermissionBoxes('channel_group', 'view_logs', true);");
   $("#translator_ide_all").attr("href", "javascript:onToggleAllPermissionBoxes('channel_group', 'translator_ide', true);");

   $("#reconfigure_none").attr("href", "javascript:onToggleAllPermissionBoxes('channel_group', 'reconfigure', false);");
   $("#startstop_none").attr("href", "javascript:onToggleAllPermissionBoxes('channel_group', 'startstop', false);");
   $("#exportlogs_none").attr("href", "javascript:onToggleAllPermissionBoxes('channel_group', 'export', false);");
   $("#view_logs_none").attr("href", "javascript:onToggleAllPermissionBoxes('channel_group', 'view_logs', false);");
   $("#translator_ide_none").attr("href", "javascript:onToggleAllPermissionBoxes('channel_group', 'translator_ide', false);");

   $("li.channel_group_row").bind("mouseover", function(){ onRowEntered($(this)); } )
                   .bind("mouseout", function(){ onRowExited(); } );

   stripChannelGroupRows();

   if(!$(".channel_group_view_state[value=visible]").length && !("editRole" == $("#action").val() && "Administrators" == $("#role").val()))
   {
      $("#viewable_channel_group_none_blurb").show();
   }
   else
   {
      $("#viewable_channel_group_none_blurb").hide();
   }

   //$("#save").attr("href", "javascript:document.getElementById('apply_changes').click();");
});



